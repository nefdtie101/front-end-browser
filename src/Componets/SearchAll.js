// This is is the search for the all category
import React from "react";
import Container from 'react-bootstrap/Container';
import Row from 'react-bootstrap/Row';



// This is a class the movie class
class SearchAll extends React.Component{
    constructor(props) {
        super(props);
        this.getAll = this.getAll.bind(this)

    }
    getAll(){
        let e = document.getElementById('searchAll').value;
        console.log(e)
        localStorage.setItem('search',e);
        window.location.replace(this.props.redirect)
    }

    render() {
        return(
            <Container fluid={true}>
                <Row>
                    <h2>{this.props.heading}</h2>
                    <input id='searchAll'/>
                    <button onClick={this.getAll}>Search</button>
                </Row>
            </Container>
        )
    }

}
export default SearchAll