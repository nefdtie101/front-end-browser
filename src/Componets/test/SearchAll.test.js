import React from 'react' ;
import SearchAll from "../SearchAll";
import renderer from 'react-test-renderer'

// This will test if the page is rendered properly
test( 'renders correctly',()=>{
    const input = renderer
        .create(
           <SearchAll/>
        )
        .toJSON();
    expect(input).toMatchSnapshot();
});