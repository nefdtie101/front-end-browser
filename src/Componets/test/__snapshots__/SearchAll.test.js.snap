// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="container-fluid"
>
  <div
    className="row"
  >
    <h2 />
    <input
      id="searchAll"
    />
    <button
      onClick={[Function]}
    >
      Search
    </button>
  </div>
</div>
`;
